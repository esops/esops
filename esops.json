 [
   "github:sartaj/esops-ts#2019-10-18-8 --subdir ts-node",
   "github:sartaj/esops-ts#2019-10-18-8 --subdir ts-ramda",
   "./package.extend.json -o ./package.json --merge"
 ]