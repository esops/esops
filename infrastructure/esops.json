{
  "stack": ["github:sartaj/typescript-node-stack/stack#0.1.0", "./"],
  "destination": "../"
}
