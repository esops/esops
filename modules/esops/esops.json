<pipe-me />
<target-react-native-web module={'@my/app-navigation'} />
<semantic-design />