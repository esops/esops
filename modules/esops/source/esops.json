[
  "npm:sartaj/typescript-target-stack",
  "http:",
  "https:",
  "git:github.com:sartaj/typescript-stack.git",
  "github:sartaj/typescript-stack"
]
