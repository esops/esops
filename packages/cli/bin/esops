#!/usr/bin/env node

require('@esops/language-babel/node')

// run the CLI with the current process arguments
require('../')(process.argv)
