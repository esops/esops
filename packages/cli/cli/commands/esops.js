// eslint-disable-next-line

const path = require('path')
const target = require('@esops/target-react-native-web').default
// const publishGHPages = require('@esops/publish-github-pages')
const log = require('@esops/logger')

const messages = require('../messages')

const devStaticGithub = async context => {
  const { parameters, cwd } = context

  switch (parameters.second) {
    case 'web':
    case 'browser':
    case 'github-pages':
      await target({ devMode: true, cwd })
      break
    case 'help':
    default:
      log.md(messages.webAvailableCommands)
  }
}

const shipStaticGithub = async context => {
  const { parameters, cwd } = context

  const distribution = {
    devMode: false,
    buildPath: path.join(cwd, './.esops/target/web/dist')
  }

  switch (parameters.second) {
    case 'web':
    case 'browser':
    case 'github-pages':
      await target(distribution)
      await publishGHPages({ ...context, ...distribution })
      break
    case 'help':
    default:
      log.md('For `esops ship`, available commands are `github-pages`.')
  }
}

// eslint-disable-next-line
module.exports = {
  run: async context => {
    const { parameters } = context

    switch (parameters.first) {
      case 'dev':
        devStaticGithub(context)
        break
      case 'ship':
        shipStaticGithub(context)
        break
      case 'help':
      case undefined:
        await log.mdFile(path.join(__dirname, '../../docs/welcome.md'))
        break
      default:
        log.md('For `esops`, available commands are `dev, ship, & help`.')
    }
  }
}
