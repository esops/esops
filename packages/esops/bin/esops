#!/usr/bin/env node


require('@esops/compiler-babel/node')

// run the CLI with the current process arguments
require('../src/cli')(process.argv)
