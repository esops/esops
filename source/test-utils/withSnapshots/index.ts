import {curry} from 'ramda'
import * as path from 'path'
import * as fs from 'fs'
import * as riteway from 'riteway'

const updateSnapshot = (name, snapshotPath, actual) => {
  fs.writeFileSync(snapshotPath, actual)
  console.log(`snapshot ${name} has been updated`)
  return {
    expected: actual,
    actual
  }
}

const toMatchSnapshot = curry((snapshotDir, name, contents) => {
  const shouldUpdateSnapshots = process.env.UPDATE_SNAPSHOTS
  const snapshotPath = path.join(snapshotDir, `${name}.snap`)

  if (!fs.existsSync(snapshotDir)) fs.mkdirSync(snapshotDir)

  const actual =
    typeof contents !== 'string' ? JSON.stringify(contents, null, 2) : contents

  if (!fs.existsSync(snapshotPath)) {
    if (shouldUpdateSnapshots) return updateSnapshot(name, snapshotPath, actual)
    else
      throw new Error(
        'Snapshot does not exist yet. Run with UPDATE_SNAPSHOTS=1 to update snapshots.'
      )
  }

  const expected = fs.readFileSync(snapshotPath, {encoding: 'utf-8'})

  if (actual !== expected) {
    if (shouldUpdateSnapshots) {
      return updateSnapshot(name, snapshotPath, actual)
    } else {
      return {
        actual,
        expected
      }
    }
  } else {
    return {
      actual,
      expected
    }
  }
})

function hashCode(str) {
  return str
    .split('')
    .reduce(
      (prevHash, currVal) =>
        ((prevHash << 5) - prevHash + currVal.charCodeAt(0)) | 0,
      0
    )
}

export const withSnapshots = curry(
  (cwd: string | null, dirname: string | null, describeMessage, callback) => {
    const dirPath: string = cwd || process.cwd()
    const snapshotDirname: string = dirname || 'snapshots'
    const snapshotDir = path.join(dirPath, snapshotDirname)
    const matchByName = toMatchSnapshot(snapshotDir)

    const describeFunc = riteway.describe
    if (typeof describeFunc === 'function') {
      describeFunc(describeMessage, async assert => {
        const assertSnap = ({given, should, snap}) => {
          const name = `${describeMessage}-${hashCode(
            describeMessage + given + should
          )}`

          assert({
            given,
            should,
            ...matchByName(name, snap)
          })
        }
        callback(assert, assertSnap)
      })
    }
  }
)
