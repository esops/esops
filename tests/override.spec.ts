import * as path from 'path'
import * as spawn from 'await-spawn'

import {Try} from 'riteway'
import * as prompts from 'prompts'
import {withSnapshots} from './test-utils/withSnapshots'
import {withTempDir} from './test-utils/withTempDir'
import {
  getSortedFilePaths,
  getFileContents,
  getJsonContents,
  cleanErrorString
} from './test-utils/fs-utils'

import esops from '../interfaces/main'
import {MOCK_STACKS} from './examples'

const describe = withSnapshots(__filename)

describe('esops1() override features', async assert => {
  await withTempDir(
    __dirname,
    MOCK_STACKS['basic-overwrite-cwd-file'],
    async cwd => {
      prompts.inject([false])
      await esops({cwd})
      await assert({
        given: 'no to override',
        should: 'not override files',
        snap: getFileContents(path.join(cwd, 'tsconfig.json'))
      })
    }
  )

  await withTempDir(
    __dirname,
    MOCK_STACKS['basic-overwrite-cwd-file'],
    async cwd => {
      prompts.inject([true])
      await esops({cwd})
      await assert({
        given: 'yes to override',
        should: 'override files',
        snap: getFileContents(path.join(cwd, 'tsconfig.json'))
      })
    }
  )

  await withTempDir(
    __dirname,
    MOCK_STACKS['basic-overwrite-cwd-file'],
    async cwd => {
      const prompts = require('prompts')
      prompts.inject([new Error()])
      await esops({cwd})
      await assert({
        given: 'canceling prompt to override',
        should: 'not override files',
        snap: getFileContents(path.join(cwd, 'tsconfig.json'))
      })
    }
  )
})
