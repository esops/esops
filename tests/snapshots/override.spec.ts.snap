module.exports[`Describe esops1() override features | Given no to override: should not override files`] = `

{
  "has-been-overwritten": false
}
`

module.exports[`Describe esops1() override features | Given yes to override: should override files`] = `

{
  "has-been-overwritten": false
}
`

module.exports[`Describe esops1() override features | Given canceling prompt to override: should not override files`] = `

{
  "has-been-overwritten": true
}
`

